src/Generator/TypeGenerator/BaseComplexType.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace Laminas\Code\Generator\TypeGenerator;
+
+abstract class BaseComplexType implements Type
+{
+    /**
+     * @param AtomicType[]|IntersectionType[] $types
+     */
+    final protected function __construct(protected array $types)
```

Comment:
Uhh, let's rather please just use union types, instead of an abstract class with `protected` fields.

Really messy to maintain this later on otherwise :)